app/controllers/api/v1/media_controller.rb

```diff
@@ -17,7 +17,7 @@ def schema_class(action)
   def index
     if has_one_assocation?
       #generate the etag here as we use the index route for linked media has_one relations
-      headers['ETag'] = gen_etag(controlled_resources)
+      headers['ETag'] = gen_etag(controlled_resources.to_a)
```

Comment:
We have to be careful about converting a large number of db scope records to an array here as the scope will not have any pagination modifiers added until it's fed through the serializer code, thus the .to_a may result in a large number of objects being pulled from db and allocated in ruby land. 

TODO: check if the `has_one_association?` is a self limiting scope modifier and thus ensures we won't have problems with large scope array conversions.

Reply:
this is safe as the `has_one` association constraint means we can only load 1 media resource into an array.